app/code/Magento/Customer/Controller/Address/FormPost.php

```diff
@@ -128,18 +134,36 @@ protected function _extractAddress()
 
     /**
      * Retrieve existing address data
+     * @param $customer
      *
      * @return array
      * @throws \Exception
      */
-    protected function getExistingAddressData()
+    protected function getExistingAddressData($customer = null)
     {
         $existingAddressData = [];
-        if ($addressId = $this->getRequest()->getParam('id')) {
+        $addressId = $this->getRequest()->getParam('id');
+
+        if ($customer) {
+            foreach ($customer->getAddresses() as $customerAddress) {
+                if ($customerAddress->getId() == $addressId) {
+                    $existingAddressData = $this->getCustomerAddressMapper()->toFlatArray($customerAddress);
+                }
+            }
+
+            if (empty($existingAddressData) && $addressId) {
+                throw new \Exception();
```

Comment:
I still think there should be a meaningful exception message like: "The address does not exist anymore"